corpus/segments.py

```diff
@@ -261,6 +262,42 @@ def run(self):
                 f.writelines(lines[start:end])
 
 
+class DynamicSplitSegmentFileJob(Job):
+
+    """
+    This split the segments to concurrent many shares.
+    It is a variant to the existing SplitSegmentFileJob, which requires a delayed variable for argument concurrent.
+    """
+
+    def __init__(self, segment_file, concurrent):
+        """
+        :param tk.Path|str segment_file: segment file
+        :param tk.Delayed concurrent: number of splits
+        """
+        self.segment_file = segment_file
+        self.concurrent = concurrent
+        self.out_split_dir = self.output_path("split", directory=True)
+
+    def tasks(self):
+        yield Task("run", resume="run", mini_task=True)
+
+    def run(self):
+        with uopen(tk.uncached_path(self.segment_file), "rt") as f:
```

Comment:
uopen will call `tk.uncached_path`for you

Reply:
true! will do